draft-davidben-tls-trust-expr.md

```diff
@@ -441,39 +441,49 @@ If a relying party receives this extension in the Certificate message, it SHOULD
+      The invariants in {{trust-store-inclusion}} imply that at most one TrustStoreInclusion will satisfy this. Implementations may evaluate this efficiently by performing a binary search over the TrustStoreInclusionList.
+
+   2. If there was no corresponding TrustStoreInclusion, continue to the next TrustExpression.
+
+   3. Otherwise, check for common values between the TrustStoreInclusion's `labels` field and the TrustExpression's `excluded_labels` field. If there are no common values, terminate this procedure and return true. The certification path matches the TrustExpressionList. Otherwise, continue to the next TrustExpression.
+
+4. If no match was found in the TrustExpressionList, return false. The certification path does not match the TrustExpressionList.
+
+When a TrustStoreInclusion's `status` is `latest_version_at_issuance`, the above procedure predicts that trust anchors in the latest version will continue to be present in later versions. This allows relying parties using later trust store versions to continue to interoperate with certification paths that predate the later version. If this prediction is incorrect and the trust anchor has been removed from the later version, {{constructing-trust-expressions}} requires that the TrustExpression include appropriate `excluded_labels` values to mitigate this.
 
 ## Subscriber Behavior
 
-Subscribers using this negotiation scheme are configured with a list of certification paths, with corresponding CertificatePropertyList ({{certificate-properties}}) structures, in some preference order. When responding to a ClientHello (as a server) or CertificateRequest (as a client) containing the `trust_expressions` extension, the subscriber collects all candidate certification paths such that all of the following are true:
+Subscribers using this negotiation scheme are configured with a list of candidate certification paths, with corresponding CertificatePropertyList ({{certificate-properties}}) structures. As a server (respectively, client) responding to a ClientHello (respectively, CertificateRequest), the subscriber selects the best candidate path, incorporating the matching algorithm described in {{evaluating-trust-expressions}}, and other criteria defined in TLS.
+
+If the subscriber selected a certification path which matched via trust expressions, it MUST include an empty `trust_expressions` extension in the first CertificateEntry.
 
-* The certification path has not expired.
+This document does not prescribe a precise selection procedure and configuration, but the following is an example implementation strategy:
```

Comment:
I think here is the place to be opinionated over serving non-expired and smallest size paths first.

Reply:
although I suspect that may come down to picking the newer of the smallest...